### PR TITLE
Fix up docs for char

### PR DESCRIPTION
**Pull request by steveklabnik**
Part of https://github.com/rust-lang/rust/issues/29333

### CODE REVIEW

File: src/libstd/primitive_docs.rs [hunk 1]
```diff
@@ -16,21 +16,87 @@ mod prim_bool { }
...
+///
+/// // five elements times one byte per element
+/// assert_eq!(5, s.len() * std::mem::size_of::<u8>());
+/// ```
+///
+/// [`String`]: string/struct.String.html
+///
+/// As always, remember that a human intutition for 'character' may not map to
+/// unicode's definitions. For example, emoji symbols such as '❤️' are more than
+/// one byte: ❤️ in particular is six:
```
Comment (bluss) [inline src/libstd/primitive_docs.rs]: sp. *intuition*. Maybe the double comma can be relieved with using `;` for the first

File: src/libstd/primitive_docs.rs [hunk 2]
```diff
@@ -16,21 +16,87 @@ mod prim_bool { }
...
+/// # Representation
+///
+/// `char` is always four bytes in size. This is a different representation than
+/// a given character would have as part of a [`String`], for example:
+///
+/// ```
+/// let v = vec!['h', 'e', 'l', 'l', 'o'];
+///
+/// // five elements times four bytes for each element
+/// assert_eq!(20, v.len() * std::mem::size_of::<char>());
```
Comment (bluss) [inline src/libstd/primitive_docs.rs]: I wonder if it is too novel or too syntactically noisy for the audience here, but rust has DST size of, `std::mem::size_of_val(&v[..])` gives 20, and it should work on a &str as well.

Comment (steveklabnik) [inline src/libstd/primitive_docs.rs]: Ah interesting. Yeah, I think I'd still prefer to keep these the way they are, but good to know.

File: src/libstd/primitive_docs.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libstd/primitive_docs.rs]: fixed, and added the ;, thanks

File: src/libstd/primitive_docs.rs (hunk 2 above)
Comment (bluss) [inline src/libstd/primitive_docs.rs]: Cool. Thanks for this, it's a good improvement to the docs.

File: src/libstd/primitive_docs.rs [hunk 3]
```diff
@@ -16,21 +16,87 @@ mod prim_bool { }
...
+/// let s = String::from("hello");
+///
+/// // five elements times one byte per element
+/// assert_eq!(5, s.len() * std::mem::size_of::<u8>());
+/// ```
+///
+/// [`String`]: string/struct.String.html
+///
+/// As always, remember that a human intuition for 'character' may not map to
+/// unicode's definitions. For example, emoji symbols such as '❤️' are more than
```
Comment (apasel422) [inline src/libstd/primitive_docs.rs]: s/unicode/Unicode/

Comment (steveklabnik) [inline src/libstd/primitive_docs.rs]: you always find one :) thanks

**Comment by Manishearth** [comment]
@bors r-

```
<stdin>:1:13: 1:15 error: character literal may only contain one codepoint: '❤
<stdin>:1 let heart = '❤️';
```

**Comment by steveklabnik** [comment]
When did that `ignore` go away? :(

**Comment by steveklabnik** [comment]
It is marked as ignore, and also, `rustdoc --test src/libstd/lib.rs` passes. So I dunno what's up here.

**Comment by apasel422** [comment]
@Manishearth Is there an issue filed for the bad span on that error message?

**Comment by steveklabnik** [comment]
Fixed a tidy error. Gonna wait and see what travis says before i re-approve though.

**Comment by Manishearth** [comment]
@apasel422 I do recall seeing some issue about emoji and stuff messing up spans before, but I don't remember what it was called. I'll look for it.

**Comment by apasel422** [comment]
@Manishearth For what it's worth, this also has a bad error message:

```rust
let x = 'ab';
```
```
error: character literal may only contain one codepoint: ';
```

**Comment by steveklabnik** [comment]
Travis hasn't passed, but a `make check` passed on my machine.

@bors: r=brson

**Comment by steveklabnik** [comment]
So it passes _tests_ but not _generation_: https://github.com/rust-lang/rust/issues/30032

**Comment by steveklabnik** [comment]
I've re-worded this so that it works now. I should start generating the docs as well as just running the doc tests :/

@bors: r=brson rollup

**Pull request merged**